planning/obstacle_stop_planner/src/node.cpp

```diff
@@ -388,26 +407,50 @@ void ObstacleStopPlannerNode::searchObstacle(
       // create one step polygon for vehicle
       createOneStepPolygon(
         p_front, p_back, one_step_move_vehicle_polygon, vehicle_info, stop_param.lateral_margin);
-      debug_ptr_->pushPolygon(
-        one_step_move_vehicle_polygon, decimate_trajectory.at(i).pose.position.z,
-        PolygonType::Vehicle);
+      if (node_param_.enable_z_axis_obstacle_filtering) {
+        debug_ptr_->pushPolyhedron(
+          one_step_move_vehicle_polygon, z_axis_min, z_axis_max, PolygonType::Vehicle);
+      } else {
+        debug_ptr_->pushPolygon(
+          one_step_move_vehicle_polygon, p_front.position.z, PolygonType::Vehicle);
+      }
 
       PointCloud::Ptr collision_pointcloud_ptr(new PointCloud);
       collision_pointcloud_ptr->header = obstacle_candidate_pointcloud_ptr->header;
 
-      planner_data.found_collision_points = withinPolygon(
-        one_step_move_vehicle_polygon, stop_param.stop_search_radius, prev_center_point,
-        next_center_point, slow_down_pointcloud_ptr, collision_pointcloud_ptr);
+      if (node_param_.enable_z_axis_obstacle_filtering) {
+        auto before_size = collision_pointcloud_ptr->size();
+        planner_data.found_collision_points = withinPolyhedron(
+          one_step_move_vehicle_polygon, stop_param.stop_search_radius, prev_center_point,
+          next_center_point, slow_down_pointcloud_ptr, collision_pointcloud_ptr, z_axis_min,
+          z_axis_max);
+        auto after_size = collision_pointcloud_ptr->size();
+        if ((after_size - before_size) > 0) {
+          auto obstacle_ros_pointcloud_debug_ptr = std::make_shared<PointCloud2>();
+          pcl::toROSMsg(*collision_pointcloud_ptr, *obstacle_ros_pointcloud_debug_ptr);
+          obstacle_ros_pointcloud_debug_ptr->header.frame_id = trajectory_header.frame_id;
+          pub_point_cloud_->publish(*obstacle_ros_pointcloud_debug_ptr);
```

Comment:
Should this code be removed ?

Reply:
@maxime-clem Thank you for your review. I leave this debug code in order to visualize the obstacle points as a point cloud. What do you think about adding a parameter enabling debugging visualization of an obstacle instead of removing it?

Reply:
The main issue currently is that the code does not build (`pub_point_cloud_` is not declared).

Otherwise it is fine to leave this code but if it is heavy to run, rather than adding parameter I recommend having a condition on the number of subscribers to the debug topic.
```C++
if(debug_pub_->get_subscription_count() + debug_pub_->get_intra_process_subscription_count() > 0) {
  /* calculate and publish debug pointclould */
}
```

Reply:
> The main issue currently is that the code does not build (pub_point_cloud_ is not declared).

I'm really sorry that the code does not build. I was checking the code several times, and I did not notice the issue. 

> Otherwise it is fine to leave this code but if it is heavy to run, rather than adding parameter I recommend having a condition on the number of subscribers to the debug topic.

Ok, It is definitely a better idea to publish point cloud depending on registered subscribers to the topic instead add another parameter. I will do it that way.